### PR TITLE
Web app bundled with API

### PR DESCRIPTION
**Pull request by dodgertodd**

### CODE REVIEW

**Review by dodgertodd — COMMENTED** [review]
This is a gigantic change that I shouldn't let live on its own any more. I did a big project restructure that I don't want to keep updating every time @kingbrendann changes the original.

File: package.json [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+        "mongoose-unique-validator": "^1.0.5",
+        "redis": "^2.7.1",
+        "request": "^2.81.0"
+    },
+    "engines": {
+        "node": "8.2.1"
+    },
+    "scripts": {
+        "start": "node server/server.js",
+        "heroku-postbuild": "cd web && npm install && npm install --only=dev --no-shrinkwrap && npm run build"
```
Comment (kingbrendann) [inline package.json:19]: What is this for?

Comment (kingbrendann) [reply package.json:19]: Oh is this to install dependencies for the web project?

Comment (dodgertodd) [reply package.json:19]: See here for an example from an official heroku guy: https://github.com/mars/heroku-cra-node

It's to get Heroku to run the express backend and a react web service using create-react-app for minimal configuration.

Comment (dodgertodd) [reply package.json:19]: https://daveceddia.com/create-react-app-express-production/

Comment (kingbrendann) [reply package.json:19]: Oh interesting. Because you need to build the react client AFTER the server is built

**Review by kingbrendann — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (dodgertodd) [reply package.json:19]: Right. Heroku is building and deploying the Express Node app, but that doesn't compile/build/webpack the source React code into /web/build, from whence the web stuff is actually served

**Pull request merged**